rpcserver.go

```diff
@@ -2000,16 +2001,38 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 			satRecv += int64(c.TotalMSatReceived.ToSatoshis())
 		}
 
-		nodePub := serverPeer.addr.IdentityKey.SerializeCompressed()
+		nodePub := serverPeer.PubKey()
+
+		syncer, ok := r.server.authGossiper.SyncManager().GossipSyncer(
+			nodePub,
+		)
+		if !ok {
+			return nil, errors.New("gossip syncer for peer not found")
```

Comment:
Are we always expected to have a gossip syncer for each connected peer?

I'm particularly thinking about the window between the peer connects, and the gossip syncer has been created. Same for disconnects. If that's the case maybe wort just logging a warning without returning an error here?

Reply:
Good point. We should always have one, but since this can be racy it's indeed better if we just log it. Fixed.